lib/actions/form.js

```diff
@@ -61,14 +61,25 @@ export function setQueryParam (payload, searchId) {
   }
 }
 
-export function parseUrlQueryString (params = getUrlParams()) {
+/**
+ * An action that parses the active URL's query params (or whatever is passed in)
+ * and updates the state with the OTP plan query params. A new search will be
+ * performed according to the behavior of setQueryParam (i.e., if the passed
+ * searchId is not null).
+ * @param  {Object} params an object containing URL query params
+ * @param  {string} source optionally contains the source of the query params
+ *                         (e.g., _CALL indicates that the source is from a past
+ *                         query made during a call taker session)
+ */
+export function parseUrlQueryString (params = getUrlParams(), source) {
   return function (dispatch, getState) {
     // Filter out the OTP (i.e. non-UI) params and set the initial query
     const planParams = {}
     Object.keys(params).forEach(key => {
       if (!key.startsWith('ui_')) planParams[key] = params[key]
     })
-    const searchId = params.ui_activeSearch || coreUtils.storage.randId()
+    let searchId = params.ui_activeSearch || coreUtils.storage.randId()
+    if (source) searchId += source
```

Comment:
This always seemed a bit hacky to me to modify an ID to append on more information. Furthermore, this method fits within a complex chain of interactions that eventually the ResponsiveWebapp ends up parsing in its componentDidUpdate method. I can't think of a better way to do it that wouldn't involve a larger refactor, so I'm fine with it for now.

Reply:
Minor grievance noted. :)